src/main/java/com/jcabi/ssh/SSH.java

```diff
@@ -215,7 +237,15 @@ protected Session session() throws IOException {
                 CharEncoding.UTF_8
             );
             jsch.setHostKeyRepository(new EasyRepo());
-            jsch.addIdentity(file.getAbsolutePath());
+            if (this.passphrase == null) {
```

Comment:
There already is SSHByPassword so I don't know if introducing new class based on encrypted/unencrypted private key is desirable as it would maybe add unnecessary complexity. 

For such a small change I find it maybe too much, but I am happy to oblige.